src/internet_identity/src/state/temp_keys.rs

```diff
@@ -1,12 +1,36 @@
 use crate::MINUTE_NS;
 use candid::Principal;
 use ic_cdk::api::time;
-use internet_identity_interface::internet_identity::types::{AnchorNumber, DeviceKey, Timestamp};
+use internet_identity_interface::internet_identity::types::{IdentityNumber, PublicKey, Timestamp};
 use std::collections::{HashMap, VecDeque};
 
 // Expiration for temp keys, the same as the front-end delegation expiry
 const TEMP_KEY_EXPIRATION_NS: u64 = 10 * MINUTE_NS;
 
+/// A temp key identifier
+#[derive(Debug, Clone, Eq, PartialOrd, PartialEq, Hash)]
+pub enum TempKeyId {
```

Comment:
Having an enum here, allows to introduce a `FlowId(String)` variant later to be able to look up `temp_key` by registration flow.